crates/aptos/src/account/list.rs

```diff
@@ -46,14 +51,26 @@ impl CliCommand<Vec<serde_json::Value>> for ListResources {
         };
 
         let client = Client::new(self.rest_options.url(&self.profile_options.profile)?);
-        let response: Vec<Resource> = client
-            .get_account_resources(account)
-            .await
-            .map_err(|err| CliError::ApiError(err.to_string()))?
-            .into_inner();
-        Ok(response
-            .iter()
-            .map(|json| json.data.clone())
-            .collect::<Vec<serde_json::Value>>())
+        let map_err_func = |err: anyhow::Error| CliError::ApiError(err.to_string());
+        let response = match self.query {
+            Some(ref q) if q == "modules" => client
```

Comment:
I'd make it lowercase, and then you can cleanup the weirdness there.
```
match self.query.map(|str| str.to_lowercase().as_str()) {
  Some("modules") => ...
  _ => ...
}

```

Actually, even better, make an `enum` for the queries.

Reply:
query is a String type that's why Some("modules") doesn't work (deref doesn't work in Some pattern matching). I added an enum to parse the command line argument into.

Reply:
Yeah, the cheat is to do a `.as_str()` we do it elsewhere :)

Reply:
Some("modules".as_str()) doesn't work though as it would complain that expression is not allowed in Some. Is there a different way to do this beside what I did?